scripted diff: fix E275 lint

willcl-ark opened the pull request:
Fix E275 errors in lint-python.py with flake8 5.0.4

After updating my venv to Python 3.6.15 and reinstalling flake8 (at the current version, 5.0.4), it now throws quite a few [E275](https://www.flake8rules.com/rules/E275.html) which are fixed with this scripted diff.

fanquake commented:
Duplicate of #26257.

MarcoFalke commented:
Not sure. To me it seems odd to enforce this whitespace but no other. If you want to enforce whitespace, it might be best to discuss for all whitespace, unless there is a reason to do it for only this one.

willcl-ark commented:
First, sorry I missed duplicating #26257 (but I also think a scripted diff is the best way to make this change).

@MarcoFalke which other whitespace did you have in mind?

IMO if we are using a specific linter then we should try to ensure those lints pass (or are disabled) and not more without adding additional lints for it. Otherwise if we change "all" whitespace today, we are likely to miss more of the same un-linted whitespace errors being introduced in the future.

MarcoFalke commented:
> scripted diff is the best way

The scripted diff as of now may modify out-of-git-tree files with the wildcard. You'd have to use `git` to get the file list. Also, it doesn't pass CI as of now.

> @MarcoFalke which other whitespace did you have in mind?

Anything you could come up with. Pretty sure there are other whitespace/formatting checks in flake. If not, you can take one from `black` as example. (Not saying you should do this, just presenting this as a reason against).

> specific linter then we should try to ensure those lints pass

Yes, the only supported version is the one installed by CI. Obviously we can't support future versions that may not have been released yet or even written.

> errors being introduced in the future

Yes, so fixing this without enforcement seems doubly-bad. (Again, not saying you should do this, just presenting this as another reason against).

MarcoFalke commented on test/functional/rpc_signrawtransactionwithkey.py:
```diff
@@ -81,7 +81,7 @@ def witness_script_test(self):
         eckey = ECKey()
         eckey.generate()
-        embedded_privkey = bytes_to_wif(eckey.get_bytes())
+        embedded_privkey = bytes_to_wif (eckey.get_bytes())
```
Pretty sure this is wrong.

maflcko reviewed (commented):
Taking a look at the number of conflict, see also my template answer:

Thank you for your contribution. While this stylistic change makes sense on its own, it comes at a cost and risk for the project as a whole. The weak motivation for the change does not justify the burden that it places on the project. A burden could be any of the following:

* Time spent on review
* Accidental introduction of bugs
* (Silent) merge conflicts, either in the branch or a backport branch. Those conflicts demand further developer and reviewer time or introduce bugs.

For more information about refactoring changes and stylistic cleanup, see
* https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#refactoring
* https://github.com/bitcoin/bitcoin/blob/master/.github/PULL_REQUEST_TEMPLATE.md
* https://github.com/bitcoin/bitcoin/issues/15465

Generally, if the style is not mentioned nor enforced by the developer notes, we leave it up to the original author to pick whatever fits them best personally and then leave it that way until the line is touched for other reasons.

Let me know if you have any questions.

willcl-ark commented:
OK thanks.

Well, I'll switch this to use git files and makes sure it passes CI and leave it as an option vs #26257.

flack commented on test/functional/feature_nulldummy.py:
```diff
@@ -72,7 +72,7 @@ def create_transaction(self, *, txid, input_details=None, addr, amount, privkey)
         eckey = ECKey()
         eckey.generate()
-        self.privkey = bytes_to_wif(eckey.get_bytes())
+        self.privkey = bytes_to_wif (eckey.get_bytes())
```
these seem to be matched by accident by the `if(` regex

willcl-ark committed:
scripted diff: fix E275 lint

-BEGIN VERIFY SCRIPT-
git grep -l "\ assert(" -- '*.py' | grep -v "src/" | xargs sed -i 's/\ assert(/\ assert\ (/g'
git grep -l "\ not(" -- '*.py' | grep -v "src/" | xargs sed -i 's/\ not(/\ not\ (/g'
git grep -l "\ if(" -- '*.py' | grep -v "src/" | xargs sed -i 's/\ if(/\ if\ (/g'
-END VERIFY SCRIPT-

willcl-ark commented:
> Thank you for your contribution. While this stylistic change makes sense on its own, it comes at a cost and risk for the project as a whole. The weak motivation for the change does not justify the burden that it places on the project.

OK.  I will just apply the patch locally as I'm using newer flake8.

MarcoFalke commented:
No objection, if you want to bump the CI version and add the rule to the excludes. Should be non-controversial.

willcl-ark committed:
di: update lint dependencies

flake8 to 5.0.4
mypy to 0.971
pyzmq to 24.0.1
vulture to 2.6

willcl-ark commented:
OK I can re-open with a bump to the CI in case the scripted diff is preferred over #26257

jonatack commented:
This seems less pythonic and at first glance I'm not sure why it would be a better solution than the PR with two ACKs. I do think it's preferable to give review feedback than open competing PRs.

MarcoFalke commented:
I said "add the rule to the excludes" should be uncontroversial, not the scripted-diff, which is still broken in many ways, as I pointed out already.

willcl-ark commented:
Ah OK I understand. I will review the PR from @jonatack which I agree, without the brackets, is more pythonic.

MarcoFalke commented:
E301, E302, E303, ... and others aren't enforced either, so why would E275 be important enough  to enforce?

willcl-ark commented:
When python was bumped to 3.6.15 I made a new venv, installed flake8 et. al. (with a simple `pip install flake8` -- not using the pinned versions from ci/lint/04_install.sh) and ran the linter, which resulted in the E275 errors that this change fixed, but no E301, E302, E303, ... errors. So I suppose newer flake8 is checking E275 by default?

I'm not sure if we expect people to manually extract and locally install pinned versions from that ci script (it is linked from the readme), add a requirements.txt to `test/`, or leave the status quo. But I suspected I would not be the only person to encounter this (and had I used GitHub's search function like a good netizen I would have seen I was correct without needing to open this conflicting PR!)
